pkg/config/defaults.go

```diff
@@ -95,7 +97,7 @@ func DefaultResource(name string, terraformSchema *schema.Resource, opts ...Reso
 		Name:              name,
 		TerraformResource: terraformSchema,
 		ShortGroup:        group,
-		Kind:              kind,
+		Kind:              typeName.CapitalizeAcronyms(kind),
```

Comment:
```suggestion
		Kind:              types.NewNameFromCamel(kind).Camel,
```
or
```suggestion
                // kind could be initialized with NewNameFromSnake
		Kind:              kind.Camel,
```

Is there an advantage over re-using the `Name` type? From the look of it, it should produce the same output.

Reply:
I've pushed another commit that assumes acronyms are always tokens separated by underscores. The previous implementation does not make that assumption, i.e., whether an acronym is a full token or it's a prefix or suffix of a token, it's processed. Thus, the previous implementation is more aggressive and assuming/processing acronyms only if they are full tokens is more conservative. 

I think we can, by default, generate more conservative names.

Reply:
I had to move the name package to get rid of an import cycle between the config and types packages. Done the renaming.